### ISSUE TITLE
Uncompilable code is generated when mapping Stream to List and using AfterMapping on result.

### ISSUE THREAD

**Issue by SimSonic** [mapstruct/mapstruct#1707]
- [x] Is this an issue (and hence not a question)? 

When mapping method accepts Stream and results to Collection/List/Set and there is an void @AfterMapping method accepting the result of mapping method (Collection/List/Set), the generated code cannot be compiled:

Source code:
```java
@Value(staticConstructor = "of")
public static final class Source {
    String text;
    int number;
}

@Data
public static class Target {
    private String text;
    private int number;
    private boolean custom;
}

@Mapper
public abstract static class Converter {

    public abstract Set<Target> convert(Stream<Source> source);

    @AfterMapping
    public void addCustomValue(@MappingTarget Set<Target> targetList) {
        targetList.forEach(t -> t.custom = true);
    }
}
```

The code produced by mapstruct:
```java
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2019-02-08T15:56:38+0700",
    comments = "version: 1.3.0.Beta2, compiler: javac, environment: Java 1.8.0_191 (Oracle Corporation)"
)
public class MapStructBugExample$ConverterImpl extends Converter {

    @Override
    public Set<Target> convert(Stream<Source> source) {
        if ( source == null ) {
            return null;
        }
        
        set.addAll( source.map( source1 -> sourceToTarget( source1 ) )
            .collect( Collectors.toCollection( HashSet<Target>::new ) )
        );

        addCustomValue( set );

        return set;
    }

    protected Target sourceToTarget(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        target.setText( source.getText() );
        target.setNumber( source.getNumber() );

        return target;
    }
}
```

In real life I would expect the generated code to be like this:
```java
    @Override
    public Set<Target> convert(Stream<Source> source) {
        if (source == null) {
            return null;
        }

        Set<Target> set = source.map(source1 -> sourceToTarget(source1))
                .collect(Collectors.toCollection(HashSet<Target>::new));

        addCustomValue(set);

        return set;
    }
```

**Comment by sjaakd** [comment]
Ok.. looks like a bug..

**Comment by sjaakd** [comment]
closed by #1707.. Will end up in 1.3
